### PR TITLE
RUMM-423 Update dd trace ot

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/java/datadog/trace/common/processor/TraceProcessor.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      }
+    }
+  }
+
+  private static boolean isEnabled(final Rule rule) {
+    boolean enabled = Config.get().isRuleEnabled(rule.getClass().getSimpleName());
+    for (final String alias : rule.aliases()) {
+      enabled &= Config.get().isRuleEnabled(alias);
+    }
+    if (!enabled) {
```
Comment (mariusc83) [inline dd-sdk-android/src/main/java/datadog/trace/common/processor/TraceProcessor.java:55]: I know that you did not added this but...what is that ??

File: dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    } else if (sampleRateValue instanceof String) {
+      try {
+        span.context()
+            .setMetric(DDTags.ANALYTICS_SAMPLE_RATE, Double.parseDouble((String) sampleRateValue));
+      } catch (final NumberFormatException ex) {
+        // ignore
+      }
+    }
+
+    if (tags.containsKey(DDTags.ANALYTICS_SAMPLE_RATE)) {
```
Comment (mariusc83) [inline dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java:37]: I wonder why don't just clean this in the if/else and they do another key check here.

File: dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  public String[] aliases() {
+    return new String[] {"AnalyticsSampleRateDecorator"};
+  }
+
+  @Override
+  public void processSpan(
+      final DDSpan span, final Map<String, Object> tags, final Collection<DDSpan> trace) {
+    final Object sampleRateValue = tags.get(DDTags.ANALYTICS_SAMPLE_RATE);
+    if (sampleRateValue instanceof Number) {
+      span.context().setMetric(DDTags.ANALYTICS_SAMPLE_RATE, (Number) sampleRateValue);
```
Comment (mariusc83) [inline dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java:27]: I am not event sure if this is for agent or for the intake :(

File: dd-sdk-android/src/main/java/datadog/trace/common/processor/TraceProcessor.java (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/java/datadog/trace/common/processor/TraceProcessor.java:55]: I think there was an SLF4J log, which is removed automatically

File: dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java (hunk 3 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java:27]: Not sure this code is used by hours :/

File: dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java (hunk 2 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/java/datadog/trace/common/processor/rule/AnalyticsSampleRateRule.java:37]: Because the tag could be added by something else

**Pull request merged**